Add option to template directory

radhus opened the pull request:
Haven't tested it with actual vault yet 🙄

radhus committed:
refactor(template): keep secrets per template run

radhus committed:
refactor(template): centralize functionality to getVaultSecret

radhus committed:
refactor(template): centralize sprig funcMap

radhus committed:
refactor(template): bring in testing.T to renderAndCompare

radhus committed:
test(template): check secrets are added to lease

radhus committed:
test(template): test RenderToFile

radhus committed:
feat(template): add RenderToDirectory

ahilsend commented on cmd/vaultify/vaultify.go:
```diff
@@ -148,6 +148,8 @@ func init() {
 	templateCmd.Flags().StringVar(&flags.templateOptions.Role, "role", "", "Vault kubernetes role to assume")
 	templateCmd.Flags().StringVar(&flags.templateOptions.TemplateFileName, "template-file", "", "Template file to render")
 	templateCmd.Flags().StringVar(&flags.templateOptions.OutputFileName, "output-file", "", "Output file")
+	templateCmd.Flags().StringVar(&flags.templateOptions.TemplateDirName, "template-dir", "", "Template diretory to render all files in")
```
```suggestion
	templateCmd.Flags().StringVar(&flags.templateOptions.TemplateDirName, "template-path", "", "Template diretory to render all files in")
```

How about we have a generic flag. And based on whether the path is a file or directory we choose what to do?

In that case we could mark the old flags as deprecated :)

radhus replied:
I had the same thought, but was afraid of deprecating. But keeping the old and just mark them deprecated could work. I'll do that!

ahilsend reviewed (dismissed):
Looks very nice! Thanks! :blush: 

I would suggest a small change to the CLI flags, rest looks good. And we need to test it.

On a side note, would be convenient to allow different auth modes, then we could simplify testing via docker-compose :thinking:

radhus committed:
feat(template): add template-path and output-path flags

Deprecated template-file and output-file as the -path flags supports
both files and directories.

Commit:
feat(run): share rendering to file/dir logic with templating

Commit:
feat(template-dir): copy symlinks from source

in kubernetes ConfigMaps files are mounted via symlinks, so this does not work otherwise

ahilsend commented:
Added 2 commits, hope that's fine with you? Tested this locally, and it works

radhus commented:
Looks good to me, merge whenever you want 👏
